### PR TITLE
Add BEAST_EXPECT, replace calls to expect

### PR DESCRIPTION
**Pull request by vinniefalco**
This helps by logging the file and line when a test postcondition is not met.

### CODE REVIEW

**Comment by willwray** [comment]
Unexpectedly, perhaps, some expects remain?
``$ git grep --name-only -P "[^a-zA-Z_]expect\s*\(" -- src/ripple/*.cpp
src/ripple/beast/container/tests/aged_associative_container.test.cpp
src/ripple/beast/core/LexicalCast.cpp
src/ripple/beast/nudb/test/beast_nudb_callgrind_test.cpp
src/ripple/beast/nudb/test/beast_nudb_recover_test.cpp
src/ripple/beast/nudb/test/beast_nudb_store_test.cpp
src/ripple/beast/nudb/test/beast_nudb_varint_test.cpp
src/ripple/beast/utility/test/beast_Zero_test.cpp
src/ripple/beast/utility/test/beast_tagged_integer_test.cpp
src/ripple/core/tests/Coroutine.test.cpp
src/ripple/core/tests/Stoppable.test.cpp
src/ripple/nodestore/tests/Database.test.cpp
src/ripple/nodestore/tests/Timing.test.cpp
src/ripple/rpc/tests/JSONRPC.test.cpp
src/ripple/rpc/tests/Status.test.cpp
src/ripple/test/jtx/impl/Env.cpp
src/ripple/test/jtx/impl/Env_test.cpp
src/ripple/test/jtx/impl/balance.cpp
src/ripple/test/jtx/impl/flags.cpp
src/ripple/test/jtx/impl/owners.cpp
```

**Comment by vinniefalco** [comment]
@willwray Some calls still remain, yes. For example, calls to `expect` from non-member functions. And the annoying `unexpected()` function. This will be addressed in a separate pull request. Some of them we can just leave alone.

File: src/ripple/nodestore/tests/Database.test.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ class Database_test : public TestBase
         // Canonicalize the source and destination batches
         std::sort (batch.begin (), batch.end (), LessThan{});
         std::sort (copy.begin (), copy.end (), LessThan{});
-        expect (areBatchesEqual (batch, copy), "Should be equal");
+        expect(areBatchesEqual (batch, copy), "Should be equal");
```
Comment (willwray) [inline src/ripple/nodestore/tests/Database.test.cpp]: Kill all strings?

Comment (willwray) [inline src/ripple/nodestore/tests/Database.test.cpp]: Was this left unchanged intentionally?

File: src/ripple/nodestore/tests/Timing.test.cpp [hunk 2]
```diff
@@ -360,7 +360,7 @@ class Timing_test : public beast::unit_test::suite
                     std::shared_ptr<NodeObject> result;
                     obj = seq1_.obj(dist_(gen_));
                     backend_.fetch(obj->getHash().data(), &result);
-                    suite_.expect (result && isSame(result, obj));
+                    suite_.expect(result && isSame(result, obj));
```
Comment (willwray) [inline src/ripple/nodestore/tests/Timing.test.cpp]: member expect comes later, right?
Also, is VS removing the space or is this an intended reformat?

File: src/ripple/nodestore/tests/Database.test.cpp (hunk 1 above)
Comment (vinniefalco) [inline src/ripple/nodestore/tests/Database.test.cpp]: Nope - willfix

File: src/ripple/nodestore/tests/Timing.test.cpp (hunk 2 above)
Comment (vinniefalco) [inline src/ripple/nodestore/tests/Timing.test.cpp]: Accidental intended reformat. I'm not sure if addressing member expect is worth it. But if it is, we can do it later.

**Comment by willwray** [comment]
:+1: 
Phew, that was a big one! With grep -> diff -> eyeball I got it covered.
All looks good. You removed lots of spelling mistakes in those comment-strings.

**Comment by miguelportilla** [comment]
:+1:

**Comment by nbougalis** [comment]
Merged as e39316882e1a37db0ed1a2e3674e780862c7b267

**Pull request closed without merging**